### PR TITLE
Enable CD

### PR DESCRIPTION
**Pull request by bzzitsme**
### Testing done

```[tasklist]
### Submitter checklist
- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

**Comment by bzzitsme** [comment]
@Vlatombe could you please review this PR?

File: pom.xml [hunk 1]
```diff
@@ -37,7 +37,7 @@
 
   <properties>
     <revision>0.12</revision>
```
Comment (Vlatombe) [inline pom.xml:39]:
```suggestion
```

**Comment by bzzitsme** [comment]
It should be OK now, if I didn't miss anything. Added: cd.yaml, updated maven.config and pom.xml
all the other things were in the plugin already (was following: https://www.jenkins.io/doc/developer/publishing/releasing-cd/)

**Comment by Vlatombe** [comment]
Still missing the dependabot entry

```
- package-ecosystem: github-actions
  directory: /
  schedule:
    interval: monthly
```

**Comment by bzzitsme** [comment]
Sorry for that, didn't notice, I got confused with https://github.com/jenkinsci/kubernetes-credentials-plugin/blob/master/.github/dependabot.yml#L3-L8 (it's for maven, but not github-actions)
will add it now

**Comment by bzzitsme** [comment]
Added in https://github.com/jenkinsci/kubernetes-credentials-plugin/pull/43/commits/780b7516222085c3cbfd724b86a0d838d5eb82f4

**Review by Vlatombe — APPROVED** [review]

**Comment by Vlatombe** [comment]
Need https://github.com/jenkins-infra/repository-permissions-updater/pull/3875 merged first.

**Comment by bzzitsme** [comment]
Thanks a lot for the review, @Vlatombe

**Comment by bzzitsme** [comment]
PR in RPU was merged, I think we can merge this PR now

**Comment by Vlatombe** [comment]
@maxlaverse FYI, and can you check https://www.jenkins.io/doc/developer/publishing/releasing-cd/#troubleshooting ? I don't have administrative right on this repo (only `write`) so can't check it.

**Comment by maxlaverse** [comment]
hi @Vlatombe ! I read the troubleshooting page. What do I need to troubleshoot ? It's meant for artifact upload errors, that would likely fail because of missing secrets, but those would be set automatically on this repository ?

**Comment by Vlatombe** [comment]
It's only to doublecheck that the CD setup happened properly and that the expected secrets are set in the repository. Otherwise we can just blindly merge and see whether the deployment happens as expected.

**Comment by maxlaverse** [comment]
Looks like the environment variables are here:
image

GitHub actions are disabled however. Does it need more permission than that ? (couldn't see it on https://www.jenkins.io/doc/developer/publishing/releasing-cd/)
image

**Comment by Vlatombe** [comment]
I'm merging, then.

**Comment by lemeurherve** [comment]
@maxlaverse I think you'll have to tick the last option and add jenkins-infra/github-reusable-workflows to the selected non-jenkinsci actions as currently the GHA is failing:
> jenkins-infra/github-reusable-workflows/.github/workflows/maven-cd.yml@v1 is not allowed to be used in jenkinsci/kubernetes-credentials-plugin. Reusable workflows in this workflow must be: within a repository owned by jenkinsci.

**Comment by maxlaverse** [comment]
I fixed the permissions issue @lemeurherve . Build is running: 
image

**Comment by maxlaverse** [comment]
Could it be that a prefix is missing for the tag ? It starts with a `-`
image

**Comment by lemeurherve** [comment]
It seems so yes, no idea how to resolve this though, sorry.

**Pull request merged**